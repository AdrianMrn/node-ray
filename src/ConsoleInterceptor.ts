import { Ray } from './Ray';

export class ConsoleInterceptor {
    public static active = false;
    public static consoleLog: any;

    public intercept() {
        ConsoleInterceptor.active = true;
        ConsoleInterceptor.consoleLog = console.log;

        console.log = this.wrapper;
    }

    public reset() {
        ConsoleInterceptor.active = false;

        console.log = ConsoleInterceptor.consoleLog;
    }

    public active(): boolean {
        return ConsoleInterceptor.active;
    }

    protected wrapper(...args: any[]) {
        if (typeof Ray.client !== 'undefined' && Ray.client.isRayAvailable()) {
            Ray.create().send(...args);
        }

        ConsoleInterceptor.consoleLog(...args);
    }
}
