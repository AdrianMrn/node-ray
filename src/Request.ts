import { Payload } from '@/Payloads/Payload';

export class Request {
    protected uuid: string;
    protected payloads: Payload[];
    protected meta: any[];

    public constructor(uuid: string, payloads: Payload[], meta: any[] = []) {
        this.uuid = uuid;
        this.payloads = payloads;
        this.meta = meta;
    }

    public toArray(): Record<string, unknown> {
        return {
            uuid: this.uuid,
            payloads: this.payloads.map(payload => payload.toArray()),
            meta: this.meta,
        };
    }

    public toJson(): string {
        return JSON.stringify(this.toArray());
    }
}
