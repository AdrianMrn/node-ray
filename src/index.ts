import { Ray, ray } from './Ray';

export { Ray, ray } from './Ray';

export default ray;
