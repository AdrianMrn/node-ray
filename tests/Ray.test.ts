/* eslint-disable no-undef */

import { FakeClient } from './TestClasses/FakeClient';
import { Ray } from './../src/Ray';

let client: FakeClient, myRay: Ray;

beforeEach(() =>
{
    client = new FakeClient();
    myRay = Ray.create(client, 'fakeUuid');
});

it('sends the ray ban payload', () =>
{
    myRay.ban();

    expect(client.sentPayloads()).toMatchSnapshot();
});

it('sends the ray charles payload', () =>
{
    myRay.charles();

    expect(client.sentPayloads()).toMatchSnapshot();
});

it('sends a color payload', () =>
{
    myRay.color('red');
    myRay.green().blue();

    expect(client.sentPayloads()).toMatchSnapshot();
});

it('sends a size payload', () =>
{
    myRay.size('lg');
    myRay.small().large();

    expect(client.sentPayloads()).toMatchSnapshot();
});

it('sends a new screen payload', () =>
{
    myRay.newScreen('test 1');

    expect(client.sentPayloads()).toMatchSnapshot();
});


it('sends a clear screen payload', () =>
{
    myRay.clearScreen();

    expect(client.sentPayloads()).toMatchSnapshot();
});

it('sends a clear all payload', () =>
{
    myRay.clearAll();

    expect(client.sentPayloads()).toMatchSnapshot();
});

it('sends an html payload', () =>
{
    myRay.html('<em>test</em>');

    expect(client.sentPayloads()).toMatchSnapshot();
});

it('sends an image payload', () =>
{
    myRay.image('http://localhost/test.png');

    expect(client.sentPayloads()).toMatchSnapshot();
});

it('sends an xml payload', () =>
{
    myRay.xml('<root><abc>1</abc></root>');

    expect(client.sentPayloads()).toMatchSnapshot();
});

it('sends a json string payload', () =>
{
    myRay.json('{"A": 123}');

    expect(client.sentPayloads()).toMatchSnapshot();
});

it('sends an object as json payload', () =>
{
    myRay.toJson(JSON.stringify({ a: 1, b: 2 }));

    expect(client.sentPayloads()).toMatchSnapshot();
});

it('sends a classname payload', () =>
{
    myRay.className(myRay);

    expect(client.sentPayloads()).toMatchSnapshot();
});

it('sends a notify payload', () =>
{
    myRay.notify('hello world');

    expect(client.sentPayloads()).toMatchSnapshot();
});

it('sends a file payload', () =>
{
    myRay.file(__dirname + '/TestData/test.txt');

    expect(client.sentPayloads()).toMatchSnapshot();
});