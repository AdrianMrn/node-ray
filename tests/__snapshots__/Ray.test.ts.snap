// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can transform a request into JSON 1`] = `"{\\"uuid\\":\\"1-2-3-4\\",\\"payloads\\":[],\\"meta\\":[{\\"test_version\\":1}]}"`;

exports[`conditionally shows a payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {},
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "remove",
      },
    ],
    "uuid": "fakeUuid",
  },
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {},
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "remove",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`doesn't blow up when calling html without a value 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "content": "",
          "label": "HTML",
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "custom",
      },
    ],
    "uuid": "fakeUuid",
  },
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "content": "",
          "label": "HTML",
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "custom",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends a boolean payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "content": true,
          "label": "Boolean",
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "custom",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends a classname payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "values": Array [
            "Ray",
          ],
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "log",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends a clear all payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {},
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "clear_all",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends a clear screen payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "name": "",
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "new_screen",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends a color payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "color": "red",
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "color",
      },
    ],
    "uuid": "fakeUuid",
  },
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "color": "green",
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "color",
      },
    ],
    "uuid": "fakeUuid",
  },
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "color": "orange",
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "color",
      },
    ],
    "uuid": "fakeUuid",
  },
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "color": "red",
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "color",
      },
    ],
    "uuid": "fakeUuid",
  },
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "color": "purple",
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "color",
      },
    ],
    "uuid": "fakeUuid",
  },
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "color": "blue",
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "color",
      },
    ],
    "uuid": "fakeUuid",
  },
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "color": "gray",
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "color",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends a custom payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "content": "test 123",
          "label": "test",
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "custom",
      },
    ],
    "uuid": "fakeUuid",
  },
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "content": "test 4",
          "label": "",
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "custom",
      },
    ],
    "uuid": "fakeUuid",
  },
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "content": "test 5",
          "label": "",
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "custom",
      },
    ],
    "uuid": "fakeUuid",
  },
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "content": "test 5",
          "label": "",
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "custom",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends a hide payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {},
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "hide",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends a json string payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "content": Object {
            "A": 123,
          },
          "label": "JSON",
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "custom",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends a new screen payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "name": "test 1",
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "new_screen",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends a notify payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "value": "hello world",
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "notify",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends a null payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "content": null,
          "label": "Null",
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "custom",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends a raw payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "values": Array [
            "one",
          ],
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "log",
      },
      Object {
        "content": Object {
          "values": Array [
            "two",
          ],
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "log",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends a remove payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {},
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "remove",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends a remove payload conditionally 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {},
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "remove",
      },
    ],
    "uuid": "fakeUuid",
  },
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {},
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "remove",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends a size payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "size": "lg",
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "size",
      },
    ],
    "uuid": "fakeUuid",
  },
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "size": "sm",
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "size",
      },
    ],
    "uuid": "fakeUuid",
  },
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "size": "lg",
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "size",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends a string payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "values": Array [
            "test",
          ],
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "log",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends a table payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "label": "table",
          "values": Array [
            1,
            2,
            "<code style=\\"font-size: 0.8rem!important;\\"><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-yellow-600 bold\\">Object</span>&nbsp;<code style=\\"font-size: 0.8rem!important;\\" class=\\"text-gray-600 \\"><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">{</span><br>&nbsp;&nbsp;&nbsp;&nbsp;<code style=\\"font-size: 0.8rem!important;\\" class=\\"bold text-green-600 p-0\\">\\"A\\"</code><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">:&nbsp;</span>1<span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">,</span><br><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">}</span></code></code>",
            "<code style=\\"font-size: 0.8rem!important;\\">Ray&nbsp;<code style=\\"font-size: 0.8rem!important;\\" class=\\"text-gray-600 \\"><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">{</span><br>&nbsp;&nbsp;&nbsp;&nbsp;<code style=\\"font-size: 0.8rem!important;\\" class=\\"bold text-green-600 p-0\\">\\"settings\\"</code><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">:&nbsp;</span>Settings&nbsp;<span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">{</span><br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<code style=\\"font-size: 0.8rem!important;\\" class=\\"bold text-green-600 p-0\\">\\"_host\\"</code><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">:&nbsp;</span><code style=\\"font-size: 0.8rem!important;\\" class=\\"bold text-green-600 p-0\\">\\"localhost\\"</code><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">,</span><br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<code style=\\"font-size: 0.8rem!important;\\" class=\\"bold text-green-600 p-0\\">\\"_port\\"</code><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">:&nbsp;</span>23510<span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">,</span><br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<code style=\\"font-size: 0.8rem!important;\\" class=\\"bold text-green-600 p-0\\">\\"always_send_raw_values\\"</code><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">:&nbsp;</span><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-indigo-600 bold\\">false</span><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">,</span><br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<code style=\\"font-size: 0.8rem!important;\\" class=\\"bold text-green-600 p-0\\">\\"enable\\"</code><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">:&nbsp;</span><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-indigo-600 bold\\">true</span><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">,</span><br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<code style=\\"font-size: 0.8rem!important;\\" class=\\"bold text-green-600 p-0\\">\\"local_path\\"</code><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">:&nbsp;</span><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-indigo-600 bold\\">null</span><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">,</span><br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<code style=\\"font-size: 0.8rem!important;\\" class=\\"bold text-green-600 p-0\\">\\"not_defined\\"</code><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">:&nbsp;</span><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-indigo-600 bold\\">false</span><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">,</span><br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<code style=\\"font-size: 0.8rem!important;\\" class=\\"bold text-green-600 p-0\\">\\"originalSettings\\"</code><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">:&nbsp;</span><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-yellow-600 bold\\">Object</span>&nbsp;<span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">{</span><br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<code style=\\"font-size: 0.8rem!important;\\" class=\\"bold text-green-600 p-0\\">\\"always_send_raw_values\\"</code><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">:&nbsp;</span><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-indigo-600 bold\\">false</span><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">,</span><br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<code style=\\"font-size: 0.8rem!important;\\" class=\\"bold text-green-600 p-0\\">\\"enable\\"</code><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">:&nbsp;</span><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-indigo-600 bold\\">true</span><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">,</span><br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<code style=\\"font-size: 0.8rem!important;\\" class=\\"bold text-green-600 p-0\\">\\"host\\"</code><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">:&nbsp;</span><code style=\\"font-size: 0.8rem!important;\\" class=\\"bold text-green-600 p-0\\">\\"localhost\\"</code><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">,</span><br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<code style=\\"font-size: 0.8rem!important;\\" class=\\"bold text-green-600 p-0\\">\\"local_path\\"</code><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">:&nbsp;</span><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-indigo-600 bold\\">null</span><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">,</span><br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<code style=\\"font-size: 0.8rem!important;\\" class=\\"bold text-green-600 p-0\\">\\"not_defined\\"</code><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">:&nbsp;</span><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-indigo-600 bold\\">false</span><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">,</span><br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<code style=\\"font-size: 0.8rem!important;\\" class=\\"bold text-green-600 p-0\\">\\"port\\"</code><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">:&nbsp;</span>23510<span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">,</span><br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<code style=\\"font-size: 0.8rem!important;\\" class=\\"bold text-green-600 p-0\\">\\"remote_path\\"</code><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">:&nbsp;</span><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-indigo-600 bold\\">null</span><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">,</span><br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">}</span><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">,</span><br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<code style=\\"font-size: 0.8rem!important;\\" class=\\"bold text-green-600 p-0\\">\\"remote_path\\"</code><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">:&nbsp;</span><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-indigo-600 bold\\">null</span><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">,</span><br>&nbsp;&nbsp;&nbsp;&nbsp;<span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">}</span><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">,</span><br>&nbsp;&nbsp;&nbsp;&nbsp;<code style=\\"font-size: 0.8rem!important;\\" class=\\"bold text-green-600 p-0\\">\\"uuid\\"</code><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">:&nbsp;</span><code style=\\"font-size: 0.8rem!important;\\" class=\\"bold text-green-600 p-0\\">\\"fakeUuid\\"</code><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">,</span><br><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">}</span></code></code>",
            true,
            null,
          ],
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "table",
      },
    ],
    "uuid": "fakeUuid",
  },
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "label": "Table",
          "values": Array [
            "<code style=\\"font-size: 0.8rem!important;\\"><code style=\\"font-size: 0.8rem!important;\\" class=\\"text-gray-500 p-0\\"><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-yellow-600 bold\\">Array</span>&nbsp;<span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">[</span><br>&nbsp;&nbsp;&nbsp;&nbsp;3<span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">,</span><br>&nbsp;&nbsp;&nbsp;&nbsp;4<span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">,</span><br><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">]</span></code></code>",
            "<code style=\\"font-size: 0.8rem!important;\\"><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-yellow-600 bold\\">Object</span>&nbsp;<code style=\\"font-size: 0.8rem!important;\\" class=\\"text-gray-600 \\"><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">{</span><br>&nbsp;&nbsp;&nbsp;&nbsp;<code style=\\"font-size: 0.8rem!important;\\" class=\\"bold text-green-600 p-0\\">\\"B\\"</code><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">:&nbsp;</span>2<span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">,</span><br><span style=\\"font-size: 0.8rem!important;\\" class=\\"text-orange-400 bold\\">}</span></code></code>",
            false,
          ],
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "table",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends an error payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "content": "<span class=\\"text-red-400 bold\\">Error</span>: <br><span class=\\"pl-5 text-gray-500\\">test error</span>",
          "label": "Error",
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "custom",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends an html payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "content": "<em>test</em>",
          "label": "HTML",
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "custom",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends an image payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "content": "<img src=\\"http://localhost/test.png\\" alt=\\"\\" />",
          "label": "Image",
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "custom",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends an object as json payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "value": "\\"{\\\\\\"a\\\\\\":1,\\\\\\"b\\\\\\":2}\\"",
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "json_string",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends an xml payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "content": "&lt;root&gt;<br>&nbsp;&nbsp;&nbsp;&nbsp;&lt;abc&gt;1&lt;/abc&gt;<br>&lt;/root&gt;",
          "label": "XML",
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "custom",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends show and hide app payloads 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {},
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "show_app",
      },
    ],
    "uuid": "fakeUuid",
  },
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {},
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "hide_app",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends the ray ban payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "values": Array [
            "🕶",
          ],
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "log",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends the ray charles payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "values": Array [
            "🎶 🎹 🎷 🕺",
          ],
        },
        "origin": Object {
          "file": "/src/Ray.ts",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "log",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;
